### ISSUE TITLE
Literate programming chapter?

### ISSUE THREAD

**Issue by AndreaSanchezTapia** [the-turing-way/the-turing-way#2549]
## Summary

- What is the name and purpose of your chapter? **Literate programming**

- Why are you proposing it? Because literate programming is a crucial tool to support open research. We could discuss tools and workflows, LaTeX and markdown but also rmarkdown, pythonbook, overleaf, and related resources. Maybe point towards reproducible manuscript submission using Latex templates and share tips of workflows (because being able to insert code in your documents sounds awesome but is not necessarily a good idea and a thing you want to do all the time). 

- What Guide is the chapter for? I'd say the Communication guide, around the section about the types of papers than can be published.

- What branch are you working on? (no branch for now but happy to create one and see if others have thought of related content, I will update the issue accordingly 😇 )

### Discussion Points

- [ ] What would be encompassed? It can start with markdown as lingua franca but build towards rmarkdown, pyhtonbook, LaTeX and Sweave... Also: workflows and folder structures when writing reproducible manuscripts.

## Who can help?

Anyone 

## Updates

- [ ] Quick discussion about the relevance and scope (_here now 2022-08-08_)
- [ ] Write chapter outline
- [ ] Add material to the chapter
- [ ] Combine materials into a readable chapter
- [ ] Proofread
- [ ] Request reviews
- [ ] Address reviews
- [ ] Merge to main branch.

**Comment by jcolomb** [comment]
is "literate programming" a synonym of "executable papers" ? I am not sure about the term meaning...

quarto and stencils seem to be the most advanced tools for that (?)

**Comment by BeastyBlacksmith** [comment]
> is "literate programming" a synonym of "executable papers" ?

Yeah, meaning of this term shifted a bit over the years from "writing code and documentation in the same document" to "writing files that contain text as well as code and being able to convert that to different output formats"

**Comment by BeastyBlacksmith** [comment]
I got some slides from a course named "Reproducible scientific computing" that I gave last year (pre quarto), that I attach here.
[LiterateProgramming.pdf](https://github.com/alan-turing-institute/the-turing-way/files/9374360/LiterateProgramming.pdf)

**Comment by AndreaSanchezTapia** [comment]
Yes, literate programming would be similar to "write texts and compile code" but about the chapter, I think it could talk in general about how to write research reports/manuscripts/documentation in a reproducible way and after a section that includes plain markdown, latex, and other text-based options (that allow version control), talk about literate programming.

**Comment by hlapp** [comment]
I have been teaching and promoting literate programming and corresponding  methodologies under the Reproducible Science umbrella (the concept of interweaving research narrative and questions with the code that implements said questions, figures and other output generated by said code, and summary and interpretation of said outputs). More generally, it's one way to create "executable papers" (e.g., [Lasser (2020)](https://doi.org/10.1038/s42005-020-00403-4)).

**Comment by aleesteele** [comment]
Hi all! Just dropping in to add in an all-contributors tag, don't mind me. I'm excited to follow this chapter as it develops, sounds like there are lots of folks that can help Andrea to scaffold a structure. Happy to review from an outsider's perspective for clarity, etc as it develops.

@all-contributors please add @AndreaSanchezTapia for ideas

**Comment by aleesteele** [comment]
@all-contributors please add @AndreaSanchezTapia for ideas

**Comment by andreaczhang** [comment]
I would like to pick up this issue!

**Comment by jhidding** [comment]
Hi, I would also like to pick up this issue. Is any one still working on this? I've done quite extensive research on literate programming and the different ways it has permeated into current day notebook interfaces. Also, I've written and maintain the [Entangled](https://entangled.github.io) package. Just to make a start:

- History of literate programming: Knuth 1984, Ramsey 1997. Emergence of notebook interfaces.
- Guiding principle: teaching fellow humans what you want the computer to do.
- Discussion of notebooks: Jupyter, what's great and what are its problems. Modern improvements: Pluto (for Julia).
- Markup based systems: RMarkdown, Quarto, Entangled, Org-mode, (maybe Typst?).
- Publication systems: Pandoc, MkDocs, MdBook, Typst.
- Challenges: maintainability, building complexity, keeping documentation up-to-date.
- Relation with other documentation types (for which I can't seem to find an existing chapter either).

**Comment by andreaczhang** [comment]
> Hi, I would also like to pick up this issue. Is any one still working on this? I've done quite extensive research on literate programming and the different ways it has permeated into current day notebook interfaces. Also, I've written and maintain the [Entangled](https://entangled.github.io) package. Just to make a start:
> 
> * History of literate programming: Knuth 1984, Ramsey 1997. Emergence of notebook interfaces.
> * Guiding principle: teaching fellow humans what you want the computer to do.
> * Discussion of notebooks: Jupyter, what's great and what are its problems. Modern improvements: Pluto (for Julia).
> * Markup based systems: RMarkdown, Quarto, Entangled, Org-mode, (maybe Typst?).
> * Publication systems: Pandoc, MkDocs, MdBook, Typst.
> * Challenges: maintainability, building complexity, keeping documentation up-to-date.
> * Relation with other documentation types (for which I can't seem to find an existing chapter either).

@jhidding Hi! Please go ahead, I'm not working on this issue. :)
